pkg/cloud/azure/provider.go

```diff
@@ -290,7 +290,6 @@ func getRetailPrice(region string, skuName string, currencyCode string, spot boo
 	retailPrice := ""
 	for _, item := range pricingPayload.Items {
 		if item.Type == "Consumption" && !strings.Contains(item.ProductName, "Windows") {
-			// if spot is true SkuName should contain "spot, if it is false it should not
 			if spot == strings.Contains(strings.ToLower(item.SkuName), " spot") {
 				retailPrice = fmt.Sprintf("%f", item.RetailPrice)
 			}
```

Comment:
```suggestion
			if  !strings.Contains(strings.ToLower(item.SkuName), " spot") {
				spotPrice = fmt.Sprintf("%f", item.RetailPrice)
			} else {
				retailPrice = fmt.Sprintf("%f", item.RetailPrice)
			}
```

Reply:
I was thinking that you could save both the ondemand and spot prices then outside of this loop do something like this.
```
if spot && spotPrice != "" {
	retailPrice = spotPrice
}
```
This way if the spot price is not found then it defaults to the ondemand price from the api

Reply:
I agree something like that would be on spot :)

Reply:
I've added something like that